src/plugins/dashboard/public/actions/duplicate_panel_action.tsx

```diff
@@ -0,0 +1,193 @@
+import { ViewMode, IContainer, PanelState } from '../embeddable_plugin';
+import { SavedObject } from '../../../saved_objects/public';
+import { DashboardPanelState, GridData } from '..';
+
+export const ACTION_DUPLICATE_PANEL = 'duplicatePanel';
+
+export interface DuplicatePanelActionContext {
+  embeddable: VisualizeEmbeddable;
+}
+
+export class DuplicatePanelAction implements ActionByType<typeof ACTION_DUPLICATE_PANEL> {
+  public readonly type = ACTION_DUPLICATE_PANEL;
+  public readonly id = ACTION_DUPLICATE_PANEL;
+  public order = 11;
+
+  constructor(private core: CoreStart) {}
+
+  public getDisplayName({ embeddable }: DuplicatePanelActionContext) {
+    if (!embeddable.getRoot() || !embeddable.getRoot().isContainer) {
+      throw new IncompatibleActionError();
+    }
+    return i18n.translate('dashboard.panel.duplicatePanel', {
+      defaultMessage: 'Duplicate panel',
+    });
+  }
+
+  public getIconType({ embeddable }: DuplicatePanelActionContext) {
+    if (!embeddable.getRoot() || !embeddable.getRoot().isContainer) {
+      throw new IncompatibleActionError();
+    }
+    return 'copy';
+  }
+
+  public async isCompatible({ embeddable }: DuplicatePanelActionContext) {
+    if (embeddable.getInput().viewMode) {
+      if (embeddable.getInput().viewMode === ViewMode.VIEW) {
+        return false;
+      }
+    }
+
+    return Boolean(embeddable.getRoot() && embeddable.getRoot().isContainer);
+  }
+
+  public async execute({ embeddable }: DuplicatePanelActionContext) {
+    if (!embeddable.getRoot() || !embeddable.getRoot().isContainer) {
+      throw new IncompatibleActionError();
+    }
+
+    const dashboard = embeddable.getRoot() as IContainer;
+    const panelToDuplicate = dashboard.getInput().panels[embeddable.id] as DashboardPanelState;
+
+    if (!panelToDuplicate || !panelToDuplicate.savedObjectId) {
+      throw new TypeError('attempt to duplicate panel without a saved object ID');
+    } else {
+      // Fetch existing saved object
+      const savedObjectToDuplicate = await this.core.savedObjects.client.get<SavedObject>(
+        embeddable.type,
+        panelToDuplicate.savedObjectId
+      );
+
+      // Create the duplicate saved object
+      const newTitle = await this.getUniqueTitle(
+        savedObjectToDuplicate.attributes.title,
+        embeddable.type
+      );
+      const duplicatedSavedObject = await this.core.savedObjects.client.create(
+        embeddable.type,
+        {
+          ..._.cloneDeep(savedObjectToDuplicate.attributes),
+          title: newTitle,
+        },
+        { references: _.cloneDeep(savedObjectToDuplicate.references) }
+      );
+
+      // Create duplicate embeddable
+      const duplicatedEmbeddable = await dashboard.addSavedObjectEmbeddable(
+        embeddable.type,
+        duplicatedSavedObject.id
+      );
+      duplicatedEmbeddable.updateInput({ vis: embeddable.getInput().vis } as VisualizeInput);
```

Comment:
This will only work for VisualizeEmbeddable type. Maybe it should be 
```
duplicatedEmbeddable.updateInput({ ...embeddable.getInput(), id: duplicatedSavedObject.id });
```

Though... I'm not sure if that id is the saved object id or the id of the embeddable.   I want to get rid of requiring "id" on input parameter in the long run.

If you want to make this specific just to visualizations sgtm, but I would move the action then into `visualize` (which I think actually just got partially migrated to NP!)